apps/hash-frontend/src/shared/table-header/bulk-actions-dropdown.tsx

```diff
@@ -0,0 +1,298 @@
+  DataTypeWithMetadata,
+  Entity,
+  EntityTypeWithMetadata,
+  PropertyTypeWithMetadata,
+} from "@local/hash-subgraph";
+import {
+  Box,
+  chipClasses,
+  ListItemIcon,
+  ListItemText,
+  Menu,
+  Tooltip,
+} from "@mui/material";
+import {
+  bindMenu,
+  bindTrigger,
+  usePopupState,
+} from "material-ui-popup-state/hooks";
+import { FunctionComponent, useCallback, useMemo } from "react";
+
+import { useArchivePage } from "../../components/hooks/use-archive-page";
+import {
+  ArchiveEntityTypeMutation,
+  ArchiveEntityTypeMutationVariables,
+  ArchivePropertyTypeMutation,
+  ArchivePropertyTypeMutationVariables,
+  UnarchiveEntityTypeMutation,
+  UnarchiveEntityTypeMutationVariables,
+  UnarchivePropertyTypeMutation,
+  UnarchivePropertyTypeMutationVariables,
+} from "../../graphql/api-types.gen";
+import {
+  archiveEntityTypeMutation,
+  unarchiveEntityTypeMutation,
+} from "../../graphql/queries/ontology/entity-type.queries";
+import {
+  archivePropertyTypeMutation,
+  unarchivePropertyTypeMutation,
+} from "../../graphql/queries/ontology/property-type.queries";
+import { useFetchEntityTypes } from "../entity-types-context/hooks";
+import { BoxArchiveIcon } from "../icons/box-archive-icon";
+import { useFetchLatestPropertyTypes } from "../latest-property-types-context";
+import { MenuItem } from "../ui";
+import {
+  isEntityPageEntity,
+  isItemArchived,
+  isType,
+  isTypeEntityType,
+  isTypePropertyType,
+} from "../util";
+
+export const BulkActionsDropdown: FunctionComponent<{
+  selectedItems: (
+    | Entity
+    | EntityTypeWithMetadata
+    | PropertyTypeWithMetadata
+    | DataTypeWithMetadata
+  )[];
+  onBulkActionCompleted?: () => void;
+}> = ({ selectedItems, onBulkActionCompleted }) => {
+  const { archivePage, unarchivePage } = useArchivePage();
+
+  const refetchEntityTypes = useFetchEntityTypes();
+
+  const [archiveEntityType] = useMutation<
+    ArchiveEntityTypeMutation,
+    ArchiveEntityTypeMutationVariables
+  >(archiveEntityTypeMutation, {
+    onCompleted: refetchEntityTypes,
+  });
+
+  const [unarchiveEntityType] = useMutation<
+    UnarchiveEntityTypeMutation,
+    UnarchiveEntityTypeMutationVariables
+  >(unarchiveEntityTypeMutation, {
+    onCompleted: refetchEntityTypes,
+  });
+
+  const refetchPropertyTypes = useFetchLatestPropertyTypes();
+
+  const [archivePropertyType] = useMutation<
+    ArchivePropertyTypeMutation,
+    ArchivePropertyTypeMutationVariables
+  >(archivePropertyTypeMutation, {
+    onCompleted: refetchPropertyTypes,
+  });
+
+  const [unarchivePropertyType] = useMutation<
+    UnarchivePropertyTypeMutation,
+    UnarchivePropertyTypeMutationVariables
+  >(unarchivePropertyTypeMutation, {
+    onCompleted: refetchPropertyTypes,
+  });
+
+  const popupState = usePopupState({
+    variant: "popover",
+    popupId: "table-header-bulk-actions-dropdown-menu",
+  });
+
+  // Whether or not the selected items can be archived or un-archived
+  const itemsAreArchiveable = useMemo(
```

Comment:
I would also check here whether the items belong either to the user or an org they are a member of.

Reply:
Good idea, have addressed in https://github.com/hashintel/hash/pull/3193/commits/fc766b9cc22219b34dfdcf06b798a6d312c9c74c